lib/libc/minimal/include/sys/stat.h

```diff
@@ -1,9 +1,51 @@
 /*
- * Copyright (c) 2018 Linaro Limited
+ * Copyright (c) 2018 Intel Corporation
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_SYS_STAT_H_
+#define ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_SYS_STAT_H_
 
-/* Dummy sys/stat.h to fulfill the requirements of certain libraries.
- */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <kernel.h>
```

Comment:
I'm with @mbolivar from the perspective of minimal libc headers should not be including kernel.h.  However I'll say it shouldn't have been put into the Zephyr's POSIX material in the first place, so that ship has sailed and though removing the include should happen it's not in scope for this PR.

Reply:
Should not be including kernel.h here

Reply:
> Should not be including kernel.h here

Again, I just moved the file around, and made minimal change to accompany that, if any. Let me try to remove that and see if it leads to cascade of unexpected CI breaks (which is very common so far with POSIX subsys stuff, and what I'm trying to fix - step by step).